### PR TITLE
Enrich catalog's description, parameters, sampleInput, and sampleOutput annotations

### PR DESCRIPTION
**Pull request by starpit**

### CODE REVIEW

File: catalog/installSystem.sh [hunk 1]
```diff
@@ -13,23 +13,51 @@ cd "$SCRIPTDIR/../bin"
...
+                                           -a parameters '[ { "name": "lines", "required": true, "type": "array", "description": "An array of strings or numbers" } ]' \
+                                           -a sampleInput '{ "lines": [4, 2, 3] }' \
+                                           -a sampleOutput '{ "lines": [4, 2, 3], "payload": "4\n2\n3" }'
+
+install utils/sort.js             util/sort -a description 'Sorts an array' \
+                                           -a parameters '[ { "name": "lines", "required": true, "type": "array", "description": "An array of strings" } ]' \
+                                           -a sampleInput '{ "lines": [4, 2, 3] }' \
+                                           -a sampleOutput '{ "lines": [2, 3, 4], "length": 3 }'
+
+install utils/head.js             util/head -a description 'Include only prefix an array' \
```
Comment (vinodmut) [inline catalog/installSystem.sh]: Replace with "Return prefix of an array"?

File: catalog/installSystem.sh [hunk 2]
```diff
@@ -13,23 +13,51 @@ cd "$SCRIPTDIR/../bin"
...
+install utils/date.js             util/date -a description 'Current date and time' \
+                                           -a sampleOutput '{ "date": "2016-03-22T00:59:55.961Z" }'
+
+install samples/hello.js          samples/helloWorld -a description 'Demonstrates logging facilities' -a parameters '[{"name": "payload", "required":false, "description":"The string to be included in the log record"}]' \
+                                           -a sampleInput '{ "payload": "Cat" }' \
+                                           -a sampleOutput '{ }' \
+                                           -a sampleLogOutput '2016-03-22T01:02:26.387624916Z stdout: hello Cat!'
+
+install samples/greeting.js       samples/greeting -a description 'Returns a friendly greeting' -a parameters '[{"name": "name", "required":false}, {"name": "place", "required":false, "description":"The string to be included in the return value"}]' \
+                                           -a sampleInput '{ "payload": "Cat", "place": "Narrowsburg" }' \
```
Comment (vinodmut) [inline catalog/installSystem.sh]: Replace "payload" with "name".

File: catalog/installSystem.sh [hunk 3]
```diff
@@ -13,23 +13,51 @@ cd "$SCRIPTDIR/../bin"
...
+                                           -a sampleOutput '{ "payload": "Hello, Cat from Narrowsburg!" }' \
```
Comment (vinodmut) [inline catalog/installSystem.sh]: Replace "payload" with "name".

File: catalog/installSystem.sh [hunk 4]
```diff
@@ -13,23 +13,51 @@ cd "$SCRIPTDIR/../bin"
...
+                                           -a sampleLogOutput "2016-03-22T01:07:08.384982272Z stdout: params: { place: 'Narrowsburg', payload: 'Cat' }"
```
Comment (vinodmut) [inline catalog/installSystem.sh]: Replace "payload" with "name".

File: catalog/installSystem.sh [hunk 5]
```diff
@@ -13,23 +13,51 @@ cd "$SCRIPTDIR/../bin"
...
+
+install samples/wc.js             samples/wordCount -a description 'Count words in a string' -a parameters '[{"name": "payload", "required":false, "description":"A string"}]'\
+                                                    -a sampleInput '{ "payload": "Five fuzzy felines"}' \
+                                                    -a sampleOutput '{ "count": 2 }' \
```
Comment (vinodmut) [inline catalog/installSystem.sh]: "count" value should be 3 for the sampleInput.

**Comment by vinodmut** [comment]
Looks good to me.

**Comment by psuter** [comment]
Rebase? Also, why is there a merge commit in there?

**Comment by starpit** [comment]
i think the merge and multiple are mostly a sign of my naiveté 

in between my first commit and my second (the second incorporated vinod's feedback), github complained about being out of sync with the master fork...

**Comment by starpit** [comment]
superseded by #110

**Pull request closed without merging**